core/src/sampling/RandomSampler.cpp

```diff
@@ -259,7 +260,22 @@ void RandomSampler::shuffle_and_split_block_no_overlap(std::vector<size_t>& samp
                                                        size_t n_all,
                                                        size_t size,
                                                        size_t block_size) {
+  std::set<size_t> heads;
+  samples.clear();
 
+  for (int i = 0; i < n_all - block_size + 1; ++i) {
+    heads.insert(i);
+  }
+  for (int i = 0; i < size; ++i) {
+    size_t head = random_number_generator() % heads.size();
+    for (int j = 0; j < block_size; ++j) {
+      // add the selected block
+      samples.push_back(head + j);
+      // delete heads causing overlap
+      heads.erase(head + j);
+      heads.erase(head - j);
+    }
+  }
```

Comment:
I have a question here. I don't quite understand this part. If we delete the heads causing overlaps in each i loop, will it influence the number of blocks we want?
I mean since i starts from 0 to size-1, will it return in total 'size' number of blocks blocks even there are blocks overlapping (of cause, we delete those overlapping blocks) with each other?

Reply:
no selected blocks are deleted. heads causing overlap are deleted so that overlapping blocks never appears in our selection.

Reply:
> no selected blocks are deleted. heads causing overlap are deleted so that overlapping blocks never appears in our selection.

Got it. Thanks!